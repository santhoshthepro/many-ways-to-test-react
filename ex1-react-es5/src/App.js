var React = require('react');
var ReactDOM = require('react-dom');
var Header = require('./Header');
var Jumbotron = require('./Jumbotron');
var CastMemberList = require('./CastMemberList');

var App = React.createClass({
  getInitialState: function() {
    return {
      castMembers: this.props.initialCastMembers,
      user: this.props.initialUser
    };
  },

  loginAction: function() {
    this.setState({ user: this.props.initialUser });;
  },

  logoutAction: function() {
    this.setState({ user: null });;
  },

  render: function() {
    return (
      <div className="container">
        <Header user={this.state.user} login={this.loginAction} logout={this.logoutAction} />
        <Jumbotron />
        <div className="panel panel-default">
          <div className="panel-heading">
            <h2 className="panel-title">Top-Billed Cast</h2>
          </div>
          <div className="panel-body">
            <CastMemberList members = {this.state.castMembers} />
          </div>
        </div>
      </div>
    );
  }
});

module.exports = App;
