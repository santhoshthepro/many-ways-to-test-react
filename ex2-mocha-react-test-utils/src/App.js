import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';
import Jumbotron from './Jumbotron';
import CastMemberList from './CastMemberList';

const App = React.createClass({
  getInitialState: function() {
    return {
      castMembers: [
        {
          id: 1,
          name: 'Rex',
          imageUrl: 'collie-yawning-a97c4a1614b53d6f55ae2464b06ba6aa.jpg',
          thumbnailUrl: 'collie-yawning-64x64.jpg',
          bio: 'Rex is an up-and-coming actor.'
        },
        {
          id: 2,
          name: 'Sally',
          imageUrl: 'collie-yawning-a97c4a1614b53d6f55ae2464b06ba6aa.jpg',
          thumbnailUrl: 'sally-night-64x64.jpg',
          bio: 'Sally is the sweetheart child star in her first grown-up role.'
        }
      ],
      user: {
        id: 1,
        username: 'gmtester'
      }
    };
  },

  render: function() {
    return (
      <div className="container">
        <Header />
        <Jumbotron />
        <div className="panel panel-default">
          <div className="panel-heading">
            <h2 className="panel-title">Top-Billed Cast</h2>
          </div>
          <div className="panel-body">
            <CastMemberList members = {this.state.castMembers} />
          </div>
        </div>
      </div>
    );
  }
});

ReactDOM.render(
  <App />,
  document.getElementById('app')
);

