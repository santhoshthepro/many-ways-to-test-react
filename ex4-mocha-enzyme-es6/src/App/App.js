import React, {PropTypes} from 'react';
import Header from '../Header/Header';
import Jumbotron from '../Jumbotron/Jumbotron';
import CastMemberList from '../Cast/CastMemberList';

class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      castMembers: this.props.initialCastMembers.slice(0, 3),
      user: this.props.initialUser
    };

    this.loginAction = this.loginAction.bind(this);
    this.logoutAction = this.logoutAction.bind(this);
  }

  loginAction() {
    this.setState({ user: this.props.initialUser });;
  }

  logoutAction() {
    this.setState({ user: null });;
  }

  render() {
    return (
      <div className="container">
        <Header user={this.state.user} login={this.loginAction} logout={this.logoutAction} />
        <Jumbotron />
        <div className="panel panel-default">
          <div className="panel-heading">
            <h2 className="panel-title">Top-Billed Cast</h2>
          </div>
          <div className="panel-body">
            <CastMemberList members={this.state.castMembers} />
          </div>
        </div>
      </div>
    );
  }
}

App.propTypes = {
  initialCastMembers: PropTypes.array.isRequired,
  initialUser: PropTypes.object
};

export default App;
