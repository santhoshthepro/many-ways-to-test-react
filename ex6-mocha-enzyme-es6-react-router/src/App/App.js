import React, {PropTypes} from 'react';
import Header from '../Header/Header';

class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      user: props.route.initialUser
    };

    this.loginAction = this.loginAction.bind(this);
    this.logoutAction = this.logoutAction.bind(this);
  }

  loginAction() {
    this.setState({
      user: this.props.route.initialUser
    });
  }

  logoutAction() {
    this.setState({ user: null });;
  }

  render() {
    return (
      <div className="container">
        <Header user={this.state.user} login={this.loginAction} logout={this.logoutAction} />
        {this.props.children}
      </div>
    );
  }
}

App.propTypes = {
  children: PropTypes.object.isRequired,
  route: PropTypes.object.isRequired

};

export default App;
