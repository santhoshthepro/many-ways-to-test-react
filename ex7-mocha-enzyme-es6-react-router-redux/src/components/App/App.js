import React, {PropTypes} from 'react';
import Header from '../Header/Header';
import {connect} from 'react-redux';

export class App extends React.Component {
  constructor(props) {
    super(props);

    this.loginAction = this.loginAction.bind(this);
    this.logoutAction = this.logoutAction.bind(this);
  }

  loginAction() {
    this.setState({
      user: this.props.route.initialUser
    });
  }

  logoutAction() {
    this.setState({ user: null });;
  }

  render() {
    return (
      <div className="container">
        <Header user={this.props.user} login={this.loginAction} logout={this.logoutAction} />
        {this.props.children}
      </div>
    );
  }
}

App.propTypes = {
  children: PropTypes.object.isRequired,
  user: PropTypes.object.isRequired

};

function mapStateToProps(state) {
  return {
    user: state.user
  };
}

export default connect(mapStateToProps)(App);
