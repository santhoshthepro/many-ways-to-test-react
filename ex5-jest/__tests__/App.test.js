import React from 'react';
import renderer from 'react-test-renderer';
import App from '../src/App/App';

describe('App', function() {
  let props;

  beforeEach(function() {
    props = {
      initialUser: {
        id: 9,
        username: 'gmtester'
      },
      initialCastMembers: [
        {
          id: 3,
          name: 'Yul Brynner',
          imageUrl: 'yul-brynner.jpg',
          thumbnailUrl: 'yul-brynne-small.jpg',
          bio: 'Exotic leading man of American films, famed as much for his completely bald head as for his performances, Yul Brynner masked much of his life in mystery and outright lies designed to tease people he considered gullible.'
        },
        {
          id: 6,
          name: 'Deborah Kerr',
          imageUrl: 'deborah-kerr.jpg',
          thumbnailUrl: 'deborah-kerr-small.jpg',
          bio: 'Deborah Kerr was born on 30 September 1921 in Helensburgh, Scotland, the daughter of Captain Arthur Kerr-Trimmer.'
        }
      ]
    };
  });

  test('App rendered static HTML', () => {
    const component = renderer.create(
      <App {...props} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
