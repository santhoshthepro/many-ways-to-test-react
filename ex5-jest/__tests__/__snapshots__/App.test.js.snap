exports[`App App rendered static HTML 1`] = `
<div
  className="container">
  <nav
    className="navbar navbar-default">
    <div
      className="container-fluid">
      <div
        className="navbar-header">
        <button
          aria-expanded="false"
          className="navbar-toggle collapsed"
          data-target="#bs-example-navbar-collapse-1"
          data-toggle="collapse"
          type="button">
          <span
            className="sr-only">
            Toggle navigation
          </span>
          <span
            className="icon-bar" />
          <span
            className="icon-bar" />
          <span
            className="icon-bar" />
        </button>
        <a
          className="navbar-brand"
          href="#">
          Dogs: The Musical
        </a>
      </div>
      <div
        className="collapse navbar-collapse"
        id="bs-example-navbar-collapse-1">
        <ul
          className="nav navbar-nav">
          <li>
            <a
              href="#">
              The Story
            </a>
          </li>
          <li>
            <a
              href="#">
              Cast
            </a>
          </li>
          <li>
            <a
              href="#">
              Director
            </a>
          </li>
          <li>
            <a
              href="#">
              Venue
            </a>
          </li>
        </ul>
        <form
          className="navbar-form navbar-right">
          <div
            className="form-group">
            <input
              className="form-control"
              placeholder="Search"
              type="text" />
          </div>
          <button
            className="btn btn-default"
            type="submit">
            Go
          </button>
        </form>
        <ul
          className="nav navbar-nav navbar-right login-block">
          <li
            className="navbar-text">
            Welcome, gmtester
          </li>
          <li>
            <a
              className="navbar-link"
              href="#"
              onClick={[Function]}>
              Logout
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="jumbotron">
    <div
      className="ghosted">
      <h1>
        Dogs: The Musical
      </h1>
      <p>
        “It’ll have you barking in your seat!”
      </p>
      <p>
        <a
          className="btn btn-primary"
          href="#">
          Buy Tickets
        </a>
      </p>
    </div>
  </div>
  <div
    className="panel panel-default">
    <div
      className="panel-heading">
      <h2
        className="panel-title">
        Top-Billed Cast
      </h2>
    </div>
    <div
      className="panel-body">
      <div
        className="member-list">
        <div
          className="media">
          <div
            className="media-left">
            <a
              href="#">
              <img
                alt="Yul Brynner"
                className="media-object"
                src="/images/yul-brynne-small.jpg" />
            </a>
          </div>
          <div
            className="media-body">
            <h2
              className="media-heading">
              Yul Brynner
            </h2>
            Exotic leading man of American films, famed as much for his completely bald head as for his performances, Yul Brynner masked much of his life in mystery and outright lies designed to tease people he considered gullible.
          </div>
        </div>
        <div
          className="media">
          <div
            className="media-left">
            <a
              href="#">
              <img
                alt="Deborah Kerr"
                className="media-object"
                src="/images/deborah-kerr-small.jpg" />
            </a>
          </div>
          <div
            className="media-body">
            <h2
              className="media-heading">
              Deborah Kerr
            </h2>
            Deborah Kerr was born on 30 September 1921 in Helensburgh, Scotland, the daughter of Captain Arthur Kerr-Trimmer.
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
